### ISSUE TITLE
display more 20px (statusbar height) when calling or some devices connecting to personal hotspot

### ISSUE THREAD

**Issue by l12ab** [tristanhimmelman/HidingNavigationBar#32]
just as the subject

**Comment by btc** [comment]
To elaborate. The navigation bar juts out into the screen when the status bar is expanded.

![img_0306](https://cloud.githubusercontent.com/assets/734339/21753349/28f6643c-d5a0-11e6-9f74-04e2d2e8c7a1.PNG)

**Comment by tristanhimmelman** [comment]
If anybody has time to add this functionality I would happily accept a PR to support this. I currently do not have time to work on this

**Comment by btc** [comment]
Unless I am misunderstanding the original post on this issue, I wouldn't consider this issue a request for additional functionality. The issue I am reporting is a bug.

Do you know what's causing the issue? If you can point me in the right direction, perhaps I could take a shot at it.

**Comment by tristanhimmelman** [comment]
@btc this feature was never actually supported by the library which is why I don't call it a bug.

I believe the root of the issue is that the ViewController changes its frame and `HidingNavigationBarManager` does not respond to this event. I would start in that class by monitoring the following events:
- `UIApplicationWillChangeStatusBarFrameNotification`
- `UIApplicationDidChangeStatusBarFrameNotification`

**Comment by btc** [comment]
Thank you. I'll have a look at this and give it a shot.

**Comment by LeoNatan** [comment]
In addition to status bar frame changes, the framework should also consider when the status bar becomes hidden and shown. The same notifications are thrown for showing / hiding the status bar, so it is possible to add all handling there.

**Comment by StefaniOSApps** [comment]
What is the current status of this problem?

![simulator screen shot - iphone 5s - 2019-01-14 at 05 53 22](https://user-images.githubusercontent.com/29314643/51098400-5fd2e480-17ca-11e9-884d-a7f55cdd13f5.png)

**Comment by StefaniOSApps** [comment]
Solution founded ! .. plz create a PullRequest

```
fileprivate func statusBarHeight() -> CGFloat {
        if UIApplication.shared.isStatusBarHidden {
            return 0
        }
        
        let statusBarSize = UIApplication.shared.statusBarFrame.size
        let extendedStatusBarDifference = abs(viewController.view.bounds.height - (UIApplication.shared.delegate?.window??.frame.size.height ?? UIScreen.main.bounds.height))
        
        return min(statusBarSize.width, statusBarSize.height) - extendedStatusBarDifference
    }
```

**Comment by l12ab** [comment]
I use another library since I submitted this issue and didnt get response.

https://github.com/andreamazz/AMScrollingNavbar

**Comment by StefaniOSApps** [comment]
Yes, I know this framework. I create my own at https://github.com/StefaniOSApps/AMScrollingNavbar - but it sucks - it use only tap gesture and too many bugs. I hope i can fix the other bugs at this framework. The current bug was fixed.

**Comment by StefaniOSApps** [comment]
I want to upgrade this Framework to iOS 11.
https://github.com/StefaniOSApps/TLYShyNavBar

Any help is welcome. Many things are already working.
